src/FieldExchanger.jl

```diff
@@ -112,41 +112,32 @@ function update_sim!(atmos_sim::Interfacer.AtmosModelSimulation, csf, turbulent_
 end
 
 """
-    update_sim!(sim::SurfaceModelSimulation, csf, area_fraction = nothing)
+    update_sim!(sim::SurfaceModelSimulation, csf, area_fraction)
 
 Updates the surface component model cache with the current coupler fields of F_turb_energy, F_radiative, F_turb_moisture, P_liq, and ρ_sfc.
 
 # Arguments
 - `sim`: [Interfacer.SurfaceModelSimulation] containing a surface model simulation object.
 - `csf`: [NamedTuple] containing coupler fields.
 """
-function update_sim!(sim::Interfacer.SurfaceModelSimulation, csf, turbulent_fluxes, area_fraction = nothing)
+function update_sim!(sim::Interfacer.SurfaceModelSimulation, csf, turbulent_fluxes, area_fraction)
     FT = eltype(area_fraction)
 
     # atmospheric surface density
     Interfacer.update_field!(sim, Val(:air_density), csf.ρ_sfc)
 
     # turbulent fluxes
+    mask = Regridder.binary_mask.(area_fraction)
```

Comment:
This probably allocates as it is called every dt. I wonder if we should have a field in cache which holds temporary calculations on the boundary space, like Atmos does. @Sbozzolo , what do you think? There will be many instances of this though, so I'd suggest leaving this for separate PR.

Reply:
Yes, good point (for a future PR)!

Reply:
I opened an issue here https://github.com/CliMA/ClimaCoupler.jl/issues/584. I can change this back to calling `binary_mask` inside the `update_field!` calls. I thought this way was more readable, but I agree we should avoid extra allocations.